hydrus/data/resource_based_classes.py

```diff
@@ -54,99 +56,143 @@ def insert_object(object_: Dict[str, Any], session: scoped_session) -> str:
     type_ = get_type(object_)
     database_class = get_database_class(type_)
     id_ = object_.get("id", None)
-    if (
-        id_ is not None and
-        session.query(exists().where(database_class.id == id_)).scalar()
-    ):
-        raise InstanceExists(type_, id_)
-    foreign_keys = database_class.__table__.foreign_keys
-    for fk in foreign_keys:
-        # the name of the column through which this foreign key relationship
-        # is being established
-        fk_column = fk.info["column_name"]
+    if collection:
+        # if type_ is of a collection class
+        members = object_['members']
+        collection_id = id_ if id_ else str(uuid.uuid4())
+        for member in members:
+            # add all the members of that collection
+            inserted_object = database_class(members=member, collection_id=collection_id)
```

Comment:
Isn't the collection supposed to have all members previously created before being added to the collection? Looks like they are being added dynamically?

Reply:
Yes, the collection is supposed to have all members previously created before being added to the collection.

Here, the `database_class` object refers to the collection table, not the instance table. So that object is for the collection table